kubernetes/generic/proxy.yaml

```diff
@@ -0,0 +1,89 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: proxy-config
+data:
+    PULSAR_MEM: "\" -Xms28g -Xmx28g -XX:MaxDirectMemorySize=28g\""
```

Comment:
I think the proxy should be able to work with lot less memory than that since is not doing any complicated logic (after the session is established, it just downgrades itself to a bare bone TCP proxy). I don't know for sure, but I would think that something in the ballpark of 4G should be enough. 

I have the plan to add at some point a script to automatically set the appropriate memory parameters for each component based on the container memory setting..

Reply:
+1 for auto-detecting the memory size on the containers. I can try 4g and see how well that works. Leaving this at default gave me a lot of connection reset errors. I wonder how many connections per second can be supported per g.

Reply:
I tested with 4g and haven't gotten any connection reset errors with the same load I was running before.

Reply:
👍 Thanks for testing it out!